sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -153,6 +153,9 @@ class ReturnHelper {
   size_t *param_value_size_ret;
 };
 
+// save discovered pi_devices & pi_platforms for quick return
+static std::vector<pi_device> piDevicesCache;
```

Comment:
Should this rather be an `std::multimap<pi_platform, pi_device>`, or even better be an `std::vector<pi_device>`  member of the `_pi_platform`, where all devices of that platform are kept. This way it is not a global (the platform's cache should probably remain global), and you don't need to filter devices of specified platform only.

Also, this shared cache access needs to be guarded with a mutex to avoid data races.

Reply:
👍 Great idea. I will work on this today. Thanks.

Reply:
Done